File: estabilizacionP1P1/julia/NavierStokes.jl
using LinearAlgebra
using Plots
plotlyjs()
using DelimitedFiles
include("lib_fem.jl")

#importo los datos de la malla
nodos = readdlm("nodes.csv",',')
elem = readdlm("elements.csv",',')
elem=Matrix{Int64}(elem)
elem = elem.+1

n_nodes = size(nodos)[1]
borde = Int64[]
for i=1:n_nodes
    x1,x2 = nodos[i,:]
    if x1==0 
        push!(borde,i)
    elseif x1==1
        push!(borde,i)
    elseif x2==1
        push!(borde,i)
    elseif x2==0
        push!(borde,i)
    end
end

#dato incial
w=zeros(2*length(nodos[:,1]))
ejemplo="Cavity"
mu=1
it = 6 #iteraciones
i=0
uh_1=solver(nodos,borde,elem,mu,w,ejemplo)
uh_2=solver(nodos,borde,elem,mu,uh_1,ejemplo)
uh_3=solver(nodos,borde,elem,mu,uh_2,ejemplo)
uh_4=solver(nodos,borde,elem,mu,uh_3,ejemplo)
uh_5=solver(nodos,borde,elem,mu,uh_4,ejemplo)

x=nodos[:,1]
y=nodos[:,2]
u1=uh_5[1:n_nodes]
u2=uh_5[n_nodes:end]

#quiver(x,y,quiver=(u1,u2))
surface(x,y,u1)

File: estabilizacionP1P1/python/NavierStokes.py
import matplotlib.pyplot as plt
from lib_fem_mixtos import *


#armo la particion y defino los elementos y nodos
N=24
mu=1
ejemplo="Cavity"
#T,nodos,elem = rect_mesh([0,1],[0,1], N, N)
#borde=Boundary(T)

#print(borde)

#plt.triplot(nodos[:,0], nodos[:,1], T.simplices)
#plt.show()


#dato incial
#w=np.zeros(2*len(T.points))


#metodo de punto fijo
iteraciones=8
i=0
while i<iteraciones:
#    uh,ph=solver(T,nodos,borde,elem,mu,w,ejemplo)
#    w=uh
    i=i+1

#X=nodos[:,0]
#Y=nodos[:,1]
#n=len(nodos[:,0])

#uh_1=uh[0:n]
#uh_2=uh[n:2*n]

#fig1, ax1 = plt.subplots()
#ax1.set_title('Solución u_h')
#Q = ax1.quiver(X, Y, uh_1, uh_2, units='width')
#qk = ax1.quiverkey(Q)

# Creating figure
#fig = plt.figure(figsize =(16, 9)) 
#ax = plt.axes(projection ='3d') 
 
# Creating color map
#my_cmap = plt.get_cmap('hot')
   
# Creating plot
#trisurf = ax.plot_trisurf(X, Y, ph,
#                         cmap = my_cmap)#,
                         #linewidth = 0.2,
                         #antialiased = True,
                         #edgecolor = 'grey') 
#fig.colorbar(trisurf, ax = ax, shrink = 0.5, aspect = 5)
#ax.set_title('Solución $p_h$')
 
# Adding labels
#ax.set_xlabel('X-axis', fontweight ='bold')
#ax.set_ylabel('Y-axis', fontweight ='bold')
#ax.set_zlabel('Z-axis', fontweight ='bold')
     
# show plot
plt.show()